libmscore/instrchange.cpp

```diff
@@ -159,6 +146,46 @@ void InstrumentChange::setupInstrument(const Instrument* instrument)
             }
       }
 
+//---------------------------------------------------------
+//   setupStaffType
+//    sets the staff type for the new instrument.
+//    returns the clef offset.
+//---------------------------------------------------------
+
+Spatium InstrumentChange::setupStaffType(Staff* staff)
+      {
+      StaffType* st = staff->staffType(segment()->measure()->tick());
+      Spatium oldOffset = st->yoffset();
+      Spatium newOffset = Spatium(2.5 - 0.5 * (float)_lines);
```

Comment:
what are these magic numbers doing here?

Reply:
It's necessary to adjust the offset of the staff when adding a staff type change to a different number of lines. I'm not sure how better to represent this, as the numbers don't really lend themselves to names for constants.

Reply:
So I'm guessing this bit is inspired/copied from StaffTypeChange? If it is, do you think it'd be feasible to avoid duplication and put the code in any one single place?

Reply:
I don't think it is present anywhere else in the project, as even when you add a staff type change you have to manually change the offsets to make it line up, if you change the number of lines. I haven't done a thorough search though.

Reply:
Oh right, sorry. The numbers should be fine to leave in in this case, although a comment wouldn't hurt.

Reply:
Oh right, sorry. The numbers should be fine to leave in in this case, although a comment wouldn't hurt.